proxy_test.go

```diff
@@ -260,7 +260,10 @@ func TestProxy(t *testing.T) {
 
 	grp.Go(func() error {
 		wg1.Wait() // sockets ready
-		proxy = zmq4.NewProxy(ctx, front, back, capt)
+		proxy, err = zmq4.NewProxy(ctx, front, back, capt)
+		if err != nil {
+			return err
+		}
 		t.Logf("proxy ready")
 		wg2.Done()
 		err := proxy.Run()
```

Comment:
this will probably fail to compile.
this shouldn't:
```suggestion
		err = proxy.Run()
```

Reply:
Added forward declaration of "err", changed := to = on proxy.Run() line.